File: other/tutorial/visualization2d.py
import sys
import os
PathPop= os.getcwd()+"/../../"

sys.path.append(PathPop+"/lib")
if (sys.platform == "win32" or sys.platform == "win64"):
    from populationpythonwin32 import * #uncomment this line for windoww
else:
    from population import * #uncomment this line for linux

#### class member ###
def classmember() :
	img = Mat2UI8(PathPop+"/image/Lena.bmp")
	img.display()#display the image
	img.display("lena",False,False)#display the image with lena title without stopping the execution without resizing the image 


#### display class ###
def classdisplay() :
	img = Mat2UI8(PathPop+"/image/Lena.bmp")
	disp =  MatNDisplay()#class to display the image
	#execute and display the result in the object window disp while this windows is not closed
	i =1
	while True :
		proc = Processing()
		img = proc.erosionRegionGrowing(img,i,2)
		img = proc.dilationRegionGrowing(img,i,2)
		i = i+1
		disp.display(img)
		if disp.is_closed() is True :
			break


#### Visualization algorithms ###
def visualizealgorithm() :
	# label image in foreground of a grey(color) imageto check segmentation or seed localization  
	img = Mat2UI8(PathPop+"/image/Lena.bmp")
	proc = Processing()
	thre = proc.thresholdOtsuMethod(img)
	visu = Visualization()
	foreground = visu.labelForeground (thre,img,0.7)
	foreground.display()
	
	# display each label with a random colour
	d = DistributionPoisson(0.001)
	field = Mat2UI32(512,512)#realisation of a discrete poisson field 
	it = field.getIteratorEDomain()
	label = 1
	while it.next() is True:
		if d.randomVariable() != 0:
			field.setValue(it.x(),label)
			label=label+1 
	proc = Processing()
	field = proc.voronoiTesselationEuclidean(field)#voronoi tesselation with  2-norm
	visu = Visualization()
	visu.labelToRandomRGB(field).display()
	
	
try:	
	classmember()
	classdisplay()
	visualizealgorithm()
except pexception, e:
	e.display()

File: other/tutorial/visualization2d.cpp
#include"Population.h"//Single header
using namespace pop;//Population namespace


//#### class member ###
void classmember(){
    Mat2UI8 img;
    img.load(POP_PROJECT_SOURCE_DIR+std::string("/image/Lena.bmp"));
    img.display();//#display the image
    img.display("lena",false,false);//#display the image with lena title without stopping the execution without resizing the image
}

//#### display class ###
void classdisplay(){
    Mat2UI8 img;
    img.load(POP_PROJECT_SOURCE_DIR+std::string("/image/Lena.bmp"));
    MatNDisplay disp;//#class to display the image
    //#execute and display the result in the object window disp while this windows is not closed
    do{
        img = Processing::erosion(img,1);
        disp.display(img);
    }while(disp.is_closed()==false);
}

//#### Visualization algorithms ###
void visualizealgorithm(){
    //# label image in foreground of a grey(color) imageto check segmentation or seed localization
    Mat2UI8 img;
    img.load(POP_PROJECT_SOURCE_DIR+std::string("/image/Lena.bmp"));
    F32 value;
    Mat2UI8 thre = Processing::thresholdOtsuMethod(img,value);
    Mat2RGBUI8 foreground = Visualization::labelForeground (thre,img,0.7);
    foreground.display();

    //# display each label with a random colour
    DistributionPoisson d(0.001);
    Mat2UI32 field(512,512);//#realisation of a discrete poisson field
    Mat2UI32::IteratorEDomain it = field.getIteratorEDomain();
    int label = 1;
    while(it.next()){
        if(d.randomVariable() != 0){
            field(it.x())=label;
            label++;
        }
    }
    Visualization::labelToRandomRGB(field).display("seed",false);
    field = Processing::voronoiTesselationEuclidean(field);//#voronoi tesselation with  2-norm
    Mat2RGBUI8 voronoi = Visualization::labelToRandomRGB(field);
    voronoi.display();

}

int main(){
    classmember();
    classdisplay();
    visualizealgorithm();
}
